docs/types.rst

```diff
@@ -266,6 +266,122 @@ to and from all integer types but implicit conversion is not allowed.
         }
     }
 
+.. index:: ! function type, ! type; function
+
+.. _function_types:
+
+Function Types
+--------------
+
+Function types can be used to assign functions to variables and passing them
+to or return them from function calls. Such variables and parameters have
+to have function types. These types come in two flavours: *internal* and *external*
+functions.
+
+Internal functions can only be used inside the current contract (more specifically,
+inside the current code unit, which also includes internal library functions
+and inherited functions) because they cannot be executed outside of the
+context of the current contract. Calling an internal function is realized
+by jumping to its entry label, just like when calling a function of the current
+contract internally.
+
+External functions consist of an address and a function signature and they can
+be passed via and returned from external function calls.
+
+Function types are notated as follows:
+
+    function (<parameter types>) {internal|external} [constant] [returns (<return types>)]
+
+In contrast to the parameter types, the return types cannot be empty - if the
+function type should not return anything, the whole ``returns (<return types>)``
+part has to be omitted.
+
+By default, function types are internal, so the ``internal`` keyword can be
```

Comment:
Ah now I see what you mean. The default for a function type is "internal". Furthermore, the default for a function implementation is also "internal", unless it is defined at contract level scope, which makes it external and internal at the same time, i.e. public.